pkg/scheduler/factory.go

```diff
@@ -512,6 +512,11 @@ func MakeDefaultErrorFunc(client clientset.Interface, podLister corelisters.PodL
 		if err := podQueue.AddUnschedulableIfNotPresent(podInfo, podQueue.SchedulingCycle()); err != nil {
 			klog.Error(err)
 		}
+
+		// Update the scheduling queue with the nominated pod information. Without
+		// this, there would be a race condition between the next scheduling cycle
+		// and the time the scheduler receives a Pod Update for the nominated pod.
+		podQueue.AddNominatedPod(podInfo.Pod, nnn)
```

Comment:
as a followup, we should just pass ```nnn``` to ```AddUnschedulableIfNotPresent```, which already invoke ```AddNominatedPod```

Reply:
Before we refactor out `MakeDefaultErrorFunc`, why not keep this line in `recordSchedulingFailure`?